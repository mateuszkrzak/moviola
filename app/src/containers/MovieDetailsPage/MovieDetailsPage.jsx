import React from 'react';

import './MovieDetailsPage.scss';
import Logo from '../../common/components/Logo/Logo';
import mockedResponse from '../../response.mock';
import MovieList from '../../common/components/MovieList/MovieList';
import MovieListTitle from './MovieListTitle/MovieListTitle';
import MovieDetails from './MovieDetails/MovieDetails';

export default class MovieDetailsPage extends React.Component {
  state = {
    data: mockedResponse.data,
  };

  render() {
    return (
      <div className="page-wrapper">
        <main className="page-header">
          <div styleName="header-wrapper">
            <Logo />
            <button styleName="search-button">Search</button>
          </div>
          <MovieDetails {...this.state.data[0]} />
        </main>
        <MovieListTitle genre={this.state.data[0].genres.join(' & ')} />
        <MovieList movies={this.state.data} />
      </div>
    );
  }
}
