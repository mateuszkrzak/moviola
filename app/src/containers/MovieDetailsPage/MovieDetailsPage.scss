@import "../../common/styles/variables.scss";

.header-wrapper {
  display: flex;
  align-items: center;
}

.search-button {
  text-transform: uppercase;
  font-weight: 600;
  border: 0;
  border-radius: 3px;
  padding: 12px 20px;
  background-color: $lightgray;
  color: $pink;
  margin-left: auto;
}
