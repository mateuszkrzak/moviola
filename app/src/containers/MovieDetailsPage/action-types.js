export const FETCH_MOVIE_BY_ID_REQUEST = 'FETCH_MOVIE_BY_ID_REQUEST';
export const FETCH_MOVIE_BY_ID_FAILURE = 'FETCH_MOVIE_BY_ID_FAILURE';
export const FETCH_MOVIE_BY_ID_SUCCESS = 'FETCH_MOVIE_BY_ID_SUCCESS';
export const FETCH_MOVIES_BY_GENRES_REQUEST = 'FETCH_MOVIES_BY_GENRES_REQUEST';
export const FETCH_MOVIES_BY_GENRES_FAILURE = 'FETCH_MOVIES_BY_GENRES_FAILURE';
export const FETCH_MOVIES_BY_GENRES_SUCCESS = 'FETCH_MOVIES_BY_GENRES_SUCCESS';
