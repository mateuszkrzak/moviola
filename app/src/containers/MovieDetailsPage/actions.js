import {
  FETCH_MOVIE_BY_ID_REQUEST,
  FETCH_MOVIE_BY_ID_FAILURE,
  FETCH_MOVIE_BY_ID_SUCCESS,
  FETCH_MOVIES_BY_GENRES_REQUEST,
  FETCH_MOVIES_BY_GENRES_FAILURE,
  FETCH_MOVIES_BY_GENRES_SUCCESS,
} from './action-types';
import moviesService from '../../services/MoviesService';

function getMovieByIdRequest() {
  return {
    type: FETCH_MOVIE_BY_ID_REQUEST,
  };
}

function getMovieByIdFailure(error) {
  return {
    type: FETCH_MOVIE_BY_ID_FAILURE,
    payload: error,
  };
}

function getMovieByIdSuccess(movie) {
  return {
    type: FETCH_MOVIE_BY_ID_SUCCESS,
    payload: movie,
  };
}

function getMoviesByGenresRequest() {
  return {
    type: FETCH_MOVIES_BY_GENRES_REQUEST,
  };
}

function getMoviesByGenresFailure(error) {
  return {
    type: FETCH_MOVIES_BY_GENRES_FAILURE,
    payload: error,
  };
}

function getMoviesByGenresSuccess(movies) {
  return {
    type: FETCH_MOVIES_BY_GENRES_SUCCESS,
    payload: movies,
  };
}

export default function getMovieByIdAndSimilarMoviesAsync(searchValue) {
  return async (dispatch, getState) => {
    dispatch(getMovieByIdRequest());
    try {
      const response = await moviesService.getMovieById(searchValue);
      dispatch(getMovieByIdSuccess(response.data));
      dispatch(getMoviesByGenresRequest());

      try {
        const { genres } = getState().details.movie;

        const similarMoviesResponse = await moviesService.getMoviesByGenres(genres);
        dispatch(getMoviesByGenresSuccess(similarMoviesResponse.data.data));
      } catch (error) {
        dispatch(getMoviesByGenresFailure(error.message));
      }
    } catch (error) {
      dispatch(getMovieByIdFailure(error.message));
    }
  };
}
