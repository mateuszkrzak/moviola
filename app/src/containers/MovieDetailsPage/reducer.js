import {
  FETCH_MOVIE_BY_ID_REQUEST,
  FETCH_MOVIE_BY_ID_FAILURE,
  FETCH_MOVIE_BY_ID_SUCCESS,
  FETCH_MOVIES_BY_GENRES_REQUEST,
  FETCH_MOVIES_BY_GENRES_FAILURE,
  FETCH_MOVIES_BY_GENRES_SUCCESS,
} from './action-types';
import createReducer from '../../redux/create-reducer';

export const initialState = {
  movie: {
    id: -1,
    title: '',
    release_date: '2018-01-30',
    vote_average: 0,
    poster_path: '',
    runtime: 0,
    overview: '',
    genres: [],
  },
  similarMovies: [],
  error: '',
  isLoading: false,
};

const fetchRequest = state => ({ ...state, isLoading: true });
const fetchFailure = (state, action) => ({
  ...state,
  isLoading: false,
  error: action,
});

const fetchMovieByIdSuccess = (state, action) => ({
  ...state,
  movie: action,
  isLoading: false,
  error: '',
});

const fetchMoviesByGenresSuccess = (state, action) => ({
  ...state,
  similarMovies: action,
  isLoading: false,
  error: '',
});

export default createReducer(initialState, {
  [FETCH_MOVIE_BY_ID_REQUEST]: fetchRequest,
  [FETCH_MOVIE_BY_ID_FAILURE]: fetchFailure,
  [FETCH_MOVIE_BY_ID_SUCCESS]: fetchMovieByIdSuccess,
  [FETCH_MOVIES_BY_GENRES_REQUEST]: fetchRequest,
  [FETCH_MOVIES_BY_GENRES_FAILURE]: fetchFailure,
  [FETCH_MOVIES_BY_GENRES_SUCCESS]: fetchMoviesByGenresSuccess,
});
