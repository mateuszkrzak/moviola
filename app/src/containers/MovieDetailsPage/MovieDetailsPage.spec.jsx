import React from 'react';
import { shallow } from 'enzyme';
import { MovieDetailsPage } from './MovieDetailsPage';

describe('MovieDetailsPage component ', () => {
  const mockMovie = {
    title: 'Test tile',
    release_date: '2020-10-20',
    vote_average: 9,
    poster_path: 'http://poster.path',
    runtime: 120,
    overview: 'Test overview',
    genres: ['Test genre', 'Another test genre'],
  };
  const getAsyncMock = jest.fn();

  test('renders MovieDetailsPage', () => {
    const props = {
      match: {
        params: {
          id: 1
        }
      },
    };
    const component = shallow(<MovieDetailsPage
      movie={mockMovie}
      similarMovies={[mockMovie, mockMovie]}
      getMovieByIdAndSimilarMoviesAsync={getAsyncMock}
      {...props}
    />);

    expect(component).toMatchSnapshot();
  });
});
