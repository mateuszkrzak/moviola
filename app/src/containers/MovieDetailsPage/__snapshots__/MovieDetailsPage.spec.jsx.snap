// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPage component  renders MovieDetailsPage 1`] = `
<div
  className="page-wrapper"
>
  <main
    className="page-header"
  >
    <div
      styleName="header-wrapper"
    >
      <Logo />
      <button
        styleName="search-button"
      >
        Search
      </button>
    </div>
    <MovieElement
      genres={
        Array [
          "Test genre",
          "Another test genre",
        ]
      }
      overview="Test overview"
      poster_path="http://poster.path"
      release_date="2020-10-20"
      runtime={120}
      title="Test tile"
      vote_average={9}
    />
  </main>
  <MovieListTitle
    genre="Test genre & Another test genre"
  />
  <MovieList
    movies={
      Array [
        Object {
          "genres": Array [
            "Test genre",
            "Another test genre",
          ],
          "overview": "Test overview",
          "poster_path": "http://poster.path",
          "release_date": "2020-10-20",
          "runtime": 120,
          "title": "Test tile",
          "vote_average": 9,
        },
        Object {
          "genres": Array [
            "Test genre",
            "Another test genre",
          ],
          "overview": "Test overview",
          "poster_path": "http://poster.path",
          "release_date": "2020-10-20",
          "runtime": 120,
          "title": "Test tile",
          "vote_average": 9,
        },
      ]
    }
  />
</div>
`;
